xocto/ranges.py

```diff
@@ -832,3 +832,9 @@ def get_finite_datetime_ranges_from_timestamps(
         # or end of the period.
         if start != end
     ]
+
+
+def period_is_fully_covered(period: Range[T], ranges: Sequence[Range[T]]) -> bool:
```

Comment:
🤔 Rather than a sequence of ranges, could we make the caller pass in something like `Range[T] | RangeSet[T]` in keeping with the other operations in the module?

Reply:
https://github.com/octoenergy/xocto/pull/69#discussion_r1186008306